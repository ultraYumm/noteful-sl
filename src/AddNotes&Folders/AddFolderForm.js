import React from "react";
import './Add.css'
import NoteContext from '../App/NoteContext';
import { withRouter } from 'react-router-dom';
import config from '../config'




class AddFolderForm extends React.Component {
  
  
  static defaultProps ={
    onAddFolder: () => {},

  }
 
  static contextType = NoteContext;
  
     
  render() {

    const noteContext = this.context
    console.log(noteContext)
 
    //const newRandomfolderid =  Math.random().toString(36).substring(10, 2)

     const onSubmitForm = (e) => {
        e.preventDefault()

        let folderToAdd =  e.target.folderToAdd.value
        console.log(folderToAdd)

      const inputValues = {
      name: folderToAdd
    }

        fetch(`${config.API_ENDPOINT}/folders/`, {
            method: 'POST',
            headers: {
              'content-type': 'application/json'
            },
            body: JSON.stringify(inputValues),
          })

          .then(res => {
            if (!res.ok) {
              // get the error message from the response,
              
              return res.json().then(error => {
                // then throw it
                throw error
              })
            }
            return res.json()
          })
          .then(() => {


          this.context.handleAddFolder(folderToAdd)
        
      })
      this.props.history.push('/')
      this.props.onAddFolder()
    }


    return (
          

          <form className = "noteList" onSubmit={onSubmitForm}>
            <h2>Create a folder</h2>
              <ul className = "inputBox">

                <li className = "inputItems">
                <label>Name</label>
              
                <input
                  name='folderToAdd'
                  type='text'
                  aria-label='name'
                  placeholder='Great'
                  required
                />
                </li>

                <li className = "inputItems"> 
                <button className= "addNoteFormButton"
              
                type= "submit"        
                >
                Add folder
              </button>
              </li>
              
              </ul>

          </form>
        )
  }
}

export default withRouter (AddFolderForm);