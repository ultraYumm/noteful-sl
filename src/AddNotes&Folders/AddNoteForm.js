import React from "react";
import './Add.css'
import NoteContext from '../App/NoteContext';
import { withRouter } from 'react-router-dom';
import config from '../config'

class AddNoteForm extends React.Component {
  
    static defaultProps ={
    onAddNote: () => {},

  }
  static contextType = NoteContext;
  
     
  render() {

    const noteContext = this.context
       
    const folderList = noteContext &&
    noteContext.folders && noteContext.folders.map((item) =>(<option>
             {item.name}</option>))

    
    const changeSelection = (value) =>{

      const folderArr = noteContext.folders.filter(num=>num.name === value)
      var folderId = folderArr[0].id  
      this.setState({ 
   
       folderId: folderId,
      
      });
    }  
        
    const onSubmitForm = (e) => {
      e.preventDefault()
     
    
    let modified = new Date();
    let note = {
        name: e.target.itemToAdd.value,
        content: e.target.contentToAdd.value,
        folderId: this.state.folderId,
        modified: modified,
      }

      fetch(`${config.API_ENDPOINT}/notes/`, {
        method: 'POST',
        headers: {
          'content-type': 'application/json'
        },
        body: JSON.stringify(note),
      })

      .then(res => {
        if (!res.ok) {
          // get the error message from the response,
          
          return res.json().then(error => {
            // then throw it
            throw error
          })
        }
        return res.json()
      })

        .then((note) => {
          this.context.handleAdd(note)
        })

        .catch(error => {
          console.error({ error })
        })
            

        this.props.onAddNote()
        this.props.history.push('/')
    }

    return (
        <form className = "noteList" onSubmit={onSubmitForm}>
         <h2>Create a note</h2>
          <ul className = "inputBox">

            <li className = "inputItems">
            <label>Name</label>
          
              <input
                name='itemToAdd'
                type='text'
                aria-label='name'
                placeholder='Panda'
              required
              />
            </li>

           <li className = "inputItems">
              <label>Content</label>
              <input
                name = 'contentToAdd'
                type='text'
                aria-label='content'
                placeholder='Black and white animal....'
              />
           </li>
  
          <li className = "inputItems">   
            <label>Folder</label>
              <select 
              id="folder"        
              onChange={e => changeSelection(e.target.value)}
              >
              <option>...</option>
              {folderList}
            </select>
         </li>
            
         <li className = "inputItems"> 
          <button 
            className= "addNoteFormButton"
            type = "submit">
            Add note
          </button>
         </li>
          
        </ul>

      </form>
        )
  }
}

export default withRouter (AddNoteForm);