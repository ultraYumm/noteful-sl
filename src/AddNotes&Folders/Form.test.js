import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddFolderForm from './AddFolderForm'
import AddNoteForm from './AddFolderForm'

describe(`NotefulForm component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  }

  it('renders a form.AddFolderForm by default', () => {
    const wrapper = shallow(<AddFolderForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the AddNoteForm given context', () => {
    const wrapper = shallow(<AddNoteForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a form.AddNoteForm by default', () => {
    const wrapper = shallow(<AddNoteForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the AddNoteForm given context', () => {
    const wrapper = shallow(<AddFolderForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
