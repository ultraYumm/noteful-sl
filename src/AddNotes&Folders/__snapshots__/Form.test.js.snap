// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotefulForm component renders a form.AddFolderForm by default 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`NotefulForm component renders a form.AddNoteForm by default 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`NotefulForm component renders the AddNoteForm given context 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`NotefulForm component renders the AddNoteForm given context 2`] = `
<Route>
  <Component />
</Route>
`;
